stdlib/src/builtin/string.mojo

```diff
@@ -2109,3 +2198,76 @@ fn _calc_format_buffer_size[type: DType]() -> Int:
         return 64 + 1
     else:
         return 128 + 1  # Add 1 for the terminator
+
+
+# ===----------------------------------------------------------------------===#
+# Format method structures
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct _FormatCurlyEntry:
+    """
+    Internally used by the `format()` method.
+
+    Purpose of the variant types:
+
+
+    - `Int` for manual indexing of *args
+
+        ```mojo
+        "{0}".format("hello")
+        ```
+
+        value field contains `0`
+
+    - `NoneType` for automatic indexing of *args
+
+        ```mojo
+        "{}".format("hello")
+        ```
+
+        value field contains `None`
+
+    `*_curly` are the indexes of the enclosing curlies
+
+    This struct only supports theses:
+
+    `{number|name|nothing}`
+
+    """
+
+    var first_curly: Int
```

Comment:
Instead of the large docstring above, could you please add simple descriptions for these fields? E.g. "the index of an opening brace around a substitution field", or something.